dvc/remote/s3.py

```diff
@@ -40,29 +40,22 @@ class RemoteS3(RemoteBase):
     PARAM_ETAG = 'etag'
 
     def __init__(self, project, config):
-        import configobj
-
         self.project = project
-        storagepath = 's3://' + config.get(Config.SECTION_AWS_STORAGEPATH,
-                                           '').lstrip('/')
+
+        storagepath = 's3://{}'.format(
+            config.get(Config.SECTION_AWS_STORAGEPATH, '').lstrip('/')
+        )
+
         self.url = config.get(Config.SECTION_REMOTE_URL, storagepath)
-        self.region = config.get(Config.SECTION_AWS_REGION, None)
-        self.profile = os.getenv('AWS_PROFILE',
-                                 config.get(Config.SECTION_AWS_PROFILE,
-                                            'default'))
-        self.endpoint_url = config.get(Config.SECTION_AWS_ENDPOINT_URL, None)
+        self.region = config.get(Config.SECTION_AWS_REGION)
+        self.profile = config.get(Config.SECTION_AWS_PROFILE)
+        self.endpoint_url = config.get(Config.SECTION_AWS_ENDPOINT_URL)
 
-        credentialpath = config.get(Config.SECTION_AWS_CREDENTIALPATH, None)
-        if credentialpath:
-            creds_conf = configobj.ConfigObj(credentialpath)
-            creds = creds_conf.get(self.profile, {})
-        else:
-            creds = self._get_aws_config_from_env()
+        # backward compatibility
+        credentialpath = config.get(Config.SECTION_AWS_CREDENTIALPATH)
```

Comment:
is it documented? do we have users for this? I would consider removing this stuff.

Reply:
may be I don't understand this ... why is it `backward compatibility`?

Reply:
Sorry, @shcheklein , didn't specify, I think that we should not support the `credentialpath`, as it can be modified using environment variables understood by `boto3` when using the default session (as we are)

Reply:
Then, should we ask the same question about `SECTION_AWS_PROFILE` or `SECTION_AWS_REGION`? I think we kept them for two reasons (and @efiop will correct me):

* to be able to commit (and eventually share) - and in this case maybe we don't need credentials path option (since it's usually local machine specific and should go into local config which is gitignored)
* to have an easy way to always override some params - this scenario is similar to using env variables, but you don't have to remember specifying them again and again